### PR TITLE
Refactor connectors 3rd party

### PR DESCRIPTION
**Pull request by erdemedeiros**
- Use constructor injection instead of field inject
- Use strategy pattern select the call strategy (with or without SLA)

### CODE REVIEW

File: trending-topic-campaigns/activiti-cloud-connectors-3rd-party/src/main/java/org/activiti/cloud/connectors/external/ShoutCallStrategy.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.cloud.connectors.external;
+
+import org.activiti.cloud.connectors.external.model.Shout;
+
+public interface ShoutCallStrategy {
```
Comment (Salaboy) [inline trending-topic-campaigns/activiti-cloud-connectors-3rd-party/src/main/java/org/activiti/cloud/connectors/external/ShoutCallStrategy.java]: @erdemedeiros I have some issues with the name: ShoutCallStrategy -> can we just call this ShoutService?

File: trending-topic-campaigns/activiti-cloud-connectors-3rd-party/src/main/java/org/activiti/cloud/connectors/external/ShoutCallStrategySelector.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * limitations under the License.
+ */
+
+package org.activiti.cloud.connectors.external;
+
+import org.activiti.cloud.connectors.external.config.SLAProperties;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ShoutCallStrategySelector {
```
Comment (Salaboy) [inline trending-topic-campaigns/activiti-cloud-connectors-3rd-party/src/main/java/org/activiti/cloud/connectors/external/ShoutCallStrategySelector.java]: This is ok as Strategy

File: trending-topic-campaigns/activiti-cloud-connectors-3rd-party/src/main/java/org/activiti/cloud/connectors/external/ShoutCallStrategy.java (hunk 1 above)
Comment (Salaboy) [reply trending-topic-campaigns/activiti-cloud-connectors-3rd-party/src/main/java/org/activiti/cloud/connectors/external/ShoutCallStrategy.java]: Or maybe ShoutServiceClient.. so we have one SLAShoutServiceClient and one Simple/Basic ShoutServiceClient

Comment (erdemedeiros) [reply trending-topic-campaigns/activiti-cloud-connectors-3rd-party/src/main/java/org/activiti/cloud/connectors/external/ShoutCallStrategy.java]: Yeah, it makes sense. I'll rename it.

File: trending-topic-campaigns/activiti-cloud-connectors-3rd-party/src/main/java/org/activiti/cloud/connectors/external/ShoutCallStrategySelector.java (hunk 2 above)
Comment (erdemedeiros) [reply trending-topic-campaigns/activiti-cloud-connectors-3rd-party/src/main/java/org/activiti/cloud/connectors/external/ShoutCallStrategySelector.java]: Or maybe I'll go with `ShoutServiceClientSelector`.

**Pull request merged**